bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -2952,8 +2952,11 @@ StyleItem[] merge (long items, int itemCount) {
 			linkBefore = false;
 		}
 		char ch = segmentsText.charAt(start);
+		if(ch == '\r') {
+			ch = segmentsText.charAt(start+1);
```

Comment:
is a boundcheck required here maybe? Just assume a single `\r` at end of line would read bejond the `end`

Reply:
yes agree

Reply:
i am checking on further if the string in full is just a "\r", what needs to be done.

Reply:
Done.